src/TorchSharp/Tensor/Tensor.cs

```diff
@@ -7377,6 +7377,12 @@ public static long max_int_value(ScalarType type)
         public static DisposeScope NewDisposeScope() =>
             DisposeScopeManager.ThreadSingleton.NewDisposeScope();
 
+        /// <summary>
+        /// Get the current dispose scope for the current thread.
+        /// </summary>
+        public static DisposeScope? CurrentDisposeScope =>
```

Comment:
5. expose the current dispose scope and add Attach method for ienumerable, which could be used with the not-disposeBatch data loaders to manually attach the tensors to the current scope